File: PY/Surtido1.py
casos = int(input(""))
casos2 = 0
gal = ""
count = 0
while(casos > 0):
    d = {}
    casos2 = int(input(""))
    while(casos2 > 0):
        gal = input("")
        if gal in d:
            d[gal] += 1
        else:
            d[gal] = 1
        casos2 -= 1
    print(min(d.values()))
    casos -= 1

File: CPP/Surtido1.cpp
#include <iostream>
#include <string>
using namespace std;
int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  int casos = 0;
  string frase = "";
  string str[500];
  cin >> casos;
  int casos2 = 0;
  for (int i = 0; i < casos; i++) {
    cin >> casos2;
    cin.ignore();
    string str[casos2];
    int seen[casos2];
    for (int j = 0; j < casos2; j++) {
      getline(cin, str[j]);
    }
    for (int l = 0; l < casos2; l++) {
      seen[l] = 0;
    }
    int count = 0;
    int small = 0;
    for (int k = 0; k < casos2; k++) {
      if (seen[k] == 0) {
        count = 0;
        for (int p = k; p < casos2; p++)
          if (str[p] == str[k]) {
            count += 1;
            seen[p] = 1;
          }
      }
      small = count;
      if (count < small)
        small = count;
    }
    cout << small << endl;
  }
}